ape_tenderly/client.py

```diff
@@ -0,0 +1,96 @@
+import os
+from typing import Any, Dict, List
+
+import requests
+from ape.exceptions import ConfigError
+from ape.utils import cached_property
+from pydantic import BaseModel, parse_obj_as
+
+TENDERLY_FORK_ID = "TENDERLY_FORK_ID"
+TENDERLY_PROJECT = "TENDERLY_PROJECT"
+TENDERLY_ACCESS_KEY = "TENDERLY_ACCESS_KEY"
+TENDERLY_GATEWAY_ACCESS_KEY = "TENDERLY_GATEWAY_ACCESS_KEY"
+
+
+class ForkDetails(BaseModel):
+    chain_config: Dict[str, Any] = {}
+
+
+class Fork(BaseModel):
+    id: str
+    network_id: int
+    block_number: int
+    details: ForkDetails
+    json_rpc_url: str
+    config: Dict[str, Any] = {}
+
+
+class TenderlyClientError(Exception):
+    pass
+
+
+class TenderlyClient:
+    @cached_property
+    def __access_key_header(self) -> Dict:
+        if not (access_key := os.environ.get(TENDERLY_ACCESS_KEY)):
+            raise ConfigError("No valid tenderly access key found.")
+
+        return {"X-Access-Key": access_key}
+
+    @cached_property
+    def _api_uri(self) -> str:
+        if not (project_name := os.environ.get(TENDERLY_PROJECT)):
+            raise ConfigError("No valid tenderly project name found.")
+
+        return f"https://api.tenderly.co/api/v2/project/{project_name}"
+
+    def get_forks(self) -> List[Fork]:
+        response = requests.get(
+            f"{self._api_uri}/forks",
+            headers=self.__access_key_header,
+        )
+
+        if not response.ok:
+            # NOTE: This will raise on any HTTP errors
+            response.raise_for_status()
+            # ...and this will raise for anything else
+            raise TenderlyClientError(f"Error processing request: {response.text}")
+
+        return parse_obj_as(List[Fork], response.json())
+
+    def create_fork(self, chain_id: int) -> Fork:
+        response = requests.post(
+            f"{self._api_uri}/forks",
+            json={
+                "name": f"ape-fork-{chain_id}",
+                "description": "Automatically created by Ape",
+                "network_id": str(chain_id),
+            },
+            headers=self.__access_key_header,
+        )
+
+        if not response.ok:
+            # NOTE: This will raise on any HTTP errors
+            response.raise_for_status()
```

Comment:
This is not necessary. Here is the implementation of `requests`. `ok()`:

```python
    @property
    def ok(self):
        """Returns True if :attr:`status_code` is less than 400, False if not.

        This attribute checks if the status code of the response is between
        400 and 600 to see if there was a client error or a server error. If
        the status code is between 200 and 400, this will return True. This
        is **not** a check to see if the response code is ``200 OK``.
        """
        try:
            self.raise_for_status()
        except HTTPError:
            return False
        return True
```

(https://requests.readthedocs.io/en/latest/_modules/requests/models/#Response.ok)

Reply:
Fixed in [47d6b21](https://github.com/ApeWorX/ape-tenderly/pull/6/commits/47d6b211709cec179de03a4f9c6457bfe3ffc947)